src/calendar/header/index.js

```diff
@@ -112,12 +112,14 @@ class CalendarHeader extends Component {
       <View>
         <View style={this.style.header}>
           {leftArrow}
-          <View style={{ flexDirection: 'row' }}>
-            <Text allowFontScaling={false} style={this.style.monthText} accessibilityTraits='header'>
-              {this.props.month.toString(this.props.monthFormat ? this.props.monthFormat : 'MMMM yyyy')}
-            </Text>
-            {indicator}
-          </View>
+          <TouchableOpacity disabled={!this.props.onPressYear} onPress={this.props.onPressYear}>
```

Comment:
Instead of the condition on the 'disabled' prop I would condition the Component used, meaning `const Component = this.props.onPressYear ? TouchableOpacity : View`.